### DISCUSSION TITLE
INVALID_FOREIGN_KEY

### DISCUSSION THREAD

**Discussion by felipelinaresgcf** [directus/directus#10829] [Archive]
When I'm trying to create a collection item, there is the next error message:
`{ "message": "Invalid foreign key in field \"id\".", "extensions": { "code": "INVALID_FOREIGN_KEY", "collection": "portfolio", "field": "id", "invalid": "7" } }`

### To Reproduce

Create an item collection

### Errors Shown

`{ "message": "Invalid foreign key in field \"id\".", "extensions": { "code": "INVALID_FOREIGN_KEY", "collection": "portfolio", "field": "id", "invalid": "7" } }`

### What version of Directus are you using?

9.4.0

### What version of Node.js are you using?

14.15.4

### What database are you using?

Postgres 13

### What browser are you using?

Chrome

### What operating system are you using?

Linux 5.4.0-62-generic

### How are you deploying Directus?

Digital Ocean

**Comment by azrikahar** [comment]
Hi, I'm having difficulty understanding what's happening here exactly, would you mind providing more detail here? Would you mind elaborating "Create an item collection", as I'm not sure how can we reproduce the error based on this.

**Comment by felipelinaresgcf** [comment]
is there any way to share with you an access to my Directus?

Because error happened just creating an item into a collection and when saving it gives me the error:

![image](https://user-images.githubusercontent.com/94630316/147928936-b09fbd62-973e-4df4-8c1e-f88e36603eeb.png)

**Comment by rijkvanzanten** [comment]
@felipelinaresgcf It looks like it's treating your primary key as a foreign key as well. What relationships do you have configured for this particular collection? It looks like there might be a many-to-one / one-to-many somewhere that's using the wrong field as it's foreign key
